atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/SimpleTransactionService.java

```diff
@@ -24,15 +24,15 @@
 import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.keyvalue.api.Value;
 import com.palantir.atlasdb.transaction.encoding.TimestampEncodingStrategy;
-import com.palantir.atlasdb.transaction.encoding.V1EncodingStrategy;
 import com.palantir.atlasdb.transaction.impl.TransactionConstants;
 
 public class SimpleTransactionService implements EncodingTransactionService {
     private final KeyValueService keyValueService;
-    private final TimestampEncodingStrategy encodingStrategy = V1EncodingStrategy.INSTANCE;
+    private final TimestampEncodingStrategy encodingStrategy;
 
-    public SimpleTransactionService(KeyValueService keyValueService) {
+    public SimpleTransactionService(KeyValueService keyValueService, TimestampEncodingStrategy encodingStrategy) {
         this.keyValueService = keyValueService;
+        this.encodingStrategy = encodingStrategy;
```

Comment:
Discussed offline - I think we probably want the table to put stuff into to be injected into the constructor as well, otherwise we need to ensure that no conflict is possible between all encoding strategies used on a given transaction table